vault/logical_system.go

```diff
@@ -265,6 +266,10 @@ func NewSystemBackend(core *Core) *SystemBackend {
 						Type:        framework.TypeCommaStringSlice,
 						Description: strings.TrimSpace(sysHelp["tune_audit_non_hmac_response_keys"][0]),
 					},
+					"listing_visibility": &framework.FieldSchema{
+						Type:        framework.TypeString,
```

Comment:
Was there a reason for not having this as a bool, something like `UnauthenticatedListingAllowed` or `AllowUnauthenticatedListing` that could default to `false`?

Reply:
The initial implementation was using `*bool` to allow for the tri-state case of auth, unauth, and unset, but after a discussion with Jeff we decided to go with using string. This would still allow for tri-state values to be used, and at the same time make it clearer what "true" and "false" meant by explicitly having those values be "auth", "unauth", and "".

Reply:
Sounds good. I had a feeling that this was thought out. Just wanted to know for sure. Makes sense 👍